tools/influx_si/influx_si.xml

```diff
@@ -1,232 +1,204 @@
-         #end for
-       #end for
-
-       influx_$si.s_i
-            $opt.noopt
-            $opt.noscale
-            #if $opt.meth != 'None':
-               $opt.meth
-            #end if
-            $opt.fullsys
-            $opt.emu
-            $opt.irand
+<tool id="influx_si" name="influx_si" version="@TOOL_VERSION@+galaxy1" profile="21.09">
+    <description>
+        Estimate metabolic fluxes and concentrations by fitting simulated labeling in metabolites to NMR/MS measurements
+    </description>
+    <creator>
+        <person givenName="Serguei" familyName="Sokol" email="sokol@insa-toulouse.fr" url="https://www.toulouse-biotechnology-institute.fr/en/plateformes-plateaux/cellule-mathematiques/" />
+        <organization name="INRAE" url="https://www.inrae.fr/" />
+        <organization name="TBI" url="https://www.toulouse-biotechnology-institute.fr/" />
+        <organization name="Mathematics Cell" url="https://www.toulouse-biotechnology-institute.fr/en/plateformes-plateaux/cellule-mathematiques/" />
+        <organization name="MetaToul-FluxoMet" url="https://www.toulouse-biotechnology-institute.fr/en/plateformes-plateaux/metatoul/" />
+        <organization name="MetaboHub2" url="https://www.metabohub.fr/" />
+    </creator>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <requirements>
+        <requirement type="package" version="@TOOL_VERSION@">influx_si</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        #for $inp in $input_main:
+            unzip $inp;
+        #end for
+               
+        ret=0;
+        influx_$si.s_i
+            $noopt
+            $noscale
+            $fullsys
+            $emu
+            $irand
+            $ln
+            $sln
+            $tikhreg
+            $lim
+            $ffguess
+            $nocalc
+            $addnoise
+            $TIMEIT
+            $prof
+            #if $opt.meth:
+              --meth='$opt.meth'
+            #end if
             #if $opt.sens:
-             --sens='$opt.sens'
+              --sens='$opt.sens'
             #end if
             #if $opt.cupx:
-             --cupx='$opt.cupx'
+              --cupx='$opt.cupx'
             #end if
             #if $opt.cupn:
-             --cupn='$opt.cupn'
+              --cupn='$opt.cupn'
             #end if
             #if $opt.cupp:
-             --cupp='$opt.cupp'
+              --cupp='$opt.cupp'
             #end if
             #if $opt.clownr:
-             --clownr='$opt.clownr'
+              --clownr='$opt.clownr'
             #end if
             #if $opt.cinout:
-             --cinout='$opt.cinout'
+              --cinout='$opt.cinout'
             #end if
             #if $opt.clowp:
-             --clowp='$opt.clowp'
+              --clowp='$opt.clowp'
             #end if
             #if $opt.np:
-             --np='$opt.np'
+              --np='$opt.np'
             #end if
```

Comment:
The number of CPU cores must not be under the users control. Use 

```suggestion
            --np='\${GALAXY_SLOTS:-1}'
```

Then Galaxy will set it automatically

Reply:
That's a thing a user should not be bothered with. For Galaxy the user does not even know where the code runs. Some Galaxy servers user HPC systems scattered over different countries. 

Bottom line is that this should be configured by the admin/galaxy. 

There is something called ressource parameters that admins can enable for some tools/users .. then the number of cores would be controllable by the user.